src/index.ts

```diff
@@ -311,13 +334,15 @@ class Geckoboard {
     this.version = version;
   }
 
-  defineDataset<T extends Fields>(schema: Schema<T>): Dataset<T> {
-    return new Dataset(schema);
-  }
-
-  async ping(): Promise<void> {
+  async request(
+    method: HTTP_METHOD,
+    path: string,
+    body?: object,
+  ): Promise<Response> {
     const auth = btoa(`${this.apiKey}:`);
-    const res = await fetch('https://api.geckoboard.com', {
+    const res = await fetch(`https://api.geckoboard.com${path}`, {
```

Comment:
Maybe this would be a good use case for `new URL(path, 'https://api.geckoboard.com')` (see https://nodejs.org/api/url.html#new-urlinput-base).

Reply:
Nice!